### PR TITLE
Adjust for Entry unification

### PR DESCRIPTION
**Pull request by danielballan**
This PR adjust for changes in intake that will be introduced in
intake/intake#490. This branch has been tested against that PR branch, as well
as intake `master` and the most recent intake release, `0.5.5`.

This is how it looks interactively against that PR branch:

```py
In [1]: import databroker                                                                                                                                                                     

In [3]: list(databroker.catalog)                                                                                                                                                              
Out[3]: 
['dev1',
 'dev2',
 'fxi_feb',
 'test1',
 'test_unpack1',
 'test_unpack3',
 'dev',
 'csx',
 'example']

In [4]: databroker.catalog.dev1                                                                                                                                                               
Out[4]: <Intake catalog: dev1>  # <--- NOT AN ENTRY ANYMORE

In [5]: type(databroker.catalog.dev1)                                                                                                                                                         
Out[5]: databroker._drivers.msgpack.BlueskyMsgpackCatalog
```

### CODE REVIEW

**Comment by danielballan** [comment]
Restarted CI to run against current intake `master` just to be sure.

**Comment by ronpandolfi** [comment]
Confirming this works on my side. Excellent work both 👍

File: databroker/core.py [hunk 1]
```diff
@@ -237,6 +250,11 @@ def get(self, **kwargs):
             self.__cache[token] = datasource
         return datasource
 
+    # def __dask_tokenize__(self):
```
Comment (gwbischof) [inline databroker/core.py:253]: I don't think we need these comments.

File: databroker/core.py [hunk 2]
```diff
@@ -245,13 +263,11 @@ class StreamEntry(Entry):
...
         return dict()
 
-    def __getstate__(self):
-        args = [arg.__getstate__() if isinstance(arg, DictSerialiseMixin)
-                else arg for arg in self._captured_init_args]
-        kwargs = OrderedDict({k: arg.__getstate__()
-                              if isinstance(arg, DictSerialiseMixin) else arg
-                              for k, arg in self._captured_init_kwargs.items()})
-        return OrderedDict(cls=self.classname, args=args, kwargs=kwargs)
+    # def __dask_tokenize__(self):
```
Comment (gwbischof) [inline databroker/core.py:266]: Same here

Comment (gwbischof) [reply databroker/core.py:266]: And a few other places also

**Comment by tacaswell** [comment]
The `__dask_tokenize__` methods are left there as a long-form TODOs / skeletons.

We played with them while sorting out how to make this fast.  One of the effects of the changes here is to defer actually computing the tokenization until actually needed (which for most cases for use is "never" because we turned off all of the persist functionality), but if we ever do need the tokenization it will be slow again.  Finishing the implementation of `__dask_tokenize__` to not convert the kwargs to a string and then hash it is the way to do that (now that upstream is changed to respect our implementation).

**Comment by gwbischof** [comment]
If we use the intake server do we need to tokenize?

**Comment by danielballan** [comment]
I think so, yes. To summarize the state of things:

Previously --- Tokenizaiton happened "greedily" when an instance was first created, so we always paid the price for it, whether the token was actually accessed or not.
In this PR --- Tokenization happens only if and when the token is first accessed. And it is cached, so any repeated access will be fast. Often, in the local use case, it is never accessed, so now things are a lot faster.
In the future --- When the `__dask_tokenize__` sketches are finished, we can make tokenization faster across the board.

**Pull request merged**